src/Idris/Parser.idr

```diff
@@ -1390,13 +1390,19 @@ topDecl fname indents
 -- collectDefs : List PDecl -> List PDecl
 collectDefs [] = []
 collectDefs (PDef annot cs :: ds)
-    = let (cs', rest) = spanBy isClause ds in
-          PDef annot (cs ++ concat cs') :: assert_total (collectDefs rest)
+    = let (csWithFC, rest) = spanBy isClause ds 
+          cs' = cs ++ concat (map snd csWithFC)
+          annot' = foldr mergeFC annot (map fst csWithFC)
+      in
+          PDef annot' cs' :: assert_total (collectDefs rest)
   where
-    isClause : PDecl -> Maybe (List PClause)
+    isClause : PDecl -> Maybe (FC, List PClause)
```

Comment:
Now that `isClause` returns exactly all of the payloads of the `PDef`
constructor, I would lift it as a toplevel function in `Idris.Syntax`
and call it `isPDef`.